hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -1082,4 +1082,25 @@ class TestCreateTable extends HoodieSparkSqlTestBase {
     }
     assertTrue(exception.getMessage.contains(s"""$tableName is not a Hudi table"""))
   }
+
+  test("Test hoodie table schema consistency for non-Avro data types") {
```

Comment:
do you also tested it with hive sync enabled?

Reply:
Yes, I have test the write operation with hive-sync. It works well because the table schema has been converted to avro types.